dev/protobuf-breaking-changes-check.sh

```diff
@@ -35,7 +35,7 @@ fi
 
 pushd sql/connect/common/src/main &&
 echo "Start protobuf breaking changes checking against $BRANCH" &&
-buf breaking --against "https://github.com/apache/spark.git#branch=$BRANCH,subdir=connector/connect/common/src/main" &&
+buf breaking --against "https://github.com/apache/spark.git#branch=$BRANCH,subdir=sql/connect/common/src/main" &&
```

Comment:
Should we perform a branch version check here? If `BRANCH` is branch-3.x, should we keep comparing it with `connector/connect/common/src/main` ?

Reply:
Do we run that against 3.5/3.4?

Reply:
branch-4.0 will against branch-3.5

Reply:
Where do we run that? I grepped through to the code base and I can't find us using it.

Reply:
It shouldn't be used in GitHub Actions, because GitHub Actions uses the following configuration: 

https://github.com/apache/spark/blob/6bbfa2dad8c70b94ca52eb7cddde5ec68efbe0b1/.github/workflows/build_and_test.yml#L752-L756

 Judging from the commit history, this seems more like a tool for local use by developers? It would be best to have @martin-g , @HyukjinKwon , or @zhengruifeng , who have modified this file, confirm it.